src/main/payment/main/G1HRPS/PaymentManager.java

```diff
@@ -152,43 +151,66 @@ public PaymentStatus GetPaymentStatus(Payment payment) {
 	}
 
 	/**
-	 * Get linked items from room service and room charges and print the list
+	 * Generates a new payment with the given arguments and prints the bill.
+	 * After printing the bill, the method returns the Payment object.
 	 * 
-	 * @param payment
+	 * @param guest_id
+	 * @param room_num
 	 * @param days_of_stay
-	 * @param rs_order_list
+	 * @param room_charge
+	 * @param discounts
+	 * @param tax
+	 * @param room_service_orders
+	 * @return	new Payment object created from given arguments.
 	 */
-	public void GenerateAndPrintBill(Payment payment, int days_of_stay, List<RoomServiceOrder> rs_order_list) {
-		
-		int room_num = payment.GetRoomNum();
-		int rso_idx = 1;
-		float rs_charges = 0;
-
-		payment.SetBillTotal();
 
+	public Payment GenerateAndPrintBill(String guest_id, int room_num, int days_of_stay, float room_charge, float discounts, float tax, List<RoomServiceOrder> room_service_orders) {
+	
+		int index;
+		boolean success;
+		float rate, init_total_bill;
+		float total_room_service_charges, cost_of_stay, price_of_order;
+
+		index = 1;
+		init_total_bill = 0;
+		success = false;
+		total_room_service_charges = 0;
+		cost_of_stay = days_of_stay * room_charge;
+		
 		System.out.println("-------Payment-------");
 		System.out.println("- Days of Stay : " + days_of_stay);
-		System.out.println("-- Cost of Stay : " + payment.GetRoomCharges());
+		System.out.printf("--- Cost of Stay ($) : %.2f\n", cost_of_stay);
 		System.out.println("- Room services ordered");
-		for(RoomServiceOrder rso : rs_order_list) {
-			if(rso.GetRoomNum() == room_num) {
-				rs_charges += rso.CalTotalPrice();
-				System.out.println("Room Service Order (" + rso_idx + ")");
-				for(MenuItem mItem : rso.GetOrderedItemList()) {
-					System.out.println(mItem);
-				}
-			}
+		for(RoomServiceOrder each_order : room_service_orders) {
+			price_of_order = each_order.CalTotalPrice();
+			System.out.println("--- Room Service Order [" + index + "]:");
+
+			// for(MenuItem menu : each_order.GetOrderedItemList()) {
```

Comment:
remove code if not needed, if it's for future implementation, then leave a comment